core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -78,24 +78,29 @@
    *
    * @param msrValue
    * @param dataType
-   * @param carbonMeasure
+   * @param scale
+   * @param precision
+   * @param dateFormat
    * @return
    */
-  public static Object getMeasureValueBasedOnDataType(String msrValue, DataType dataType,
-      int scale, int precision) {
-    return getMeasureValueBasedOnDataType(msrValue, dataType, scale, precision, false);
+  public static Object getMeasureValueBasedOnDataType(String msrValue, DataType dataType, int scale,
+      int precision, String dateFormat) {
```

Comment:
format not related measure. Better add another method to convert timestamp to long and call directly